### PR TITLE
Fix .editorconfig in project root is not considered as tasks input.

### LINKED ISSUE #209: .editorconfig change seems to be ignored
I am not sure if this is Ktlintgradle bug or ktlint itself.

I have a mono repo with multiple projects. `.editirconfig` exists in the root of mono repo, but The Android project does not have it's own `.editirconfig` file:

Monorepo root
- `.editirconfig`
    - `android`
    - `iOs`

it looks like changes to `.editirconfig` are ignored (due to cache?). 

Steps 
1. Run ktlint gradle inside Android project `./gradlew ktlintcheck` that should pass
2. Make change in the `.editirconfig` that should make ktlint check fail eg. add `indent_size = 2`
3. Run ktlint 

Result
All checks are OK

Expected results
Some checks should fail

I discovered that by editing one of the files in the Android project (let's say added unwanted space) the check will start failing, so this leads to a conclusion that maybe cache is not invalidated when the `.editirconfig` file is updated.

### PR DESCRIPTION
**Pull request by Tapchicoma**
This prevents task to reset UP-TO-DATE state on any changes to this file.

Fixes #209

### CODE REVIEW

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt [hunk 1]
```diff
@@ -57,15 +57,15 @@ private tailrec fun searchEditorConfigFiles(
     fileCollection: FileCollection
 ): FileCollection {
     val editorConfigFC = projectPath.resolve(EDITOR_CONFIG_FILE_NAME)
-    val outputCollection = if (editorConfigFC != null) {
+    val outputCollection = if (editorConfigFC.toFile().exists()) {
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Is this evaluated eagerly or at task execution time?

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: I think eagerly on task configuration phase

Comment (JLLeitschuh) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Is that a problem? Do we care if the file is added/modified by a previous task?

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: I would say - we care, see, for example, #210 discussion.

What is the best way to do it? Current implementation: https://github.com/JLLeitschuh/ktlint-gradle/blob/0b019e0c0d99c2a34c2ff40fef61a1fefc21eebf/plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt#L42
 
Use `by lazy { .. }` at this line or use `ObjectFactory.property { .. }`?

Comment (JLLeitschuh) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Make it an input to the task that's a `FileProperty`? Honestly, I'd have to dig into where this code is actually called and see why you're adding it to this particular `FileCollection` to understand better what's going on.

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: I think using `by lazy { ... }` will work here.

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/PluginUtil.kt]: Tested, works good - `getEditorconfigFiles()` is only evaluated on task execution when Gradle gets inputs for the task.

File: plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt [hunk 2]
```diff
@@ -39,7 +39,9 @@ open class KtlintCheckTask @Inject constructor(
 
     @get:PathSensitive(PathSensitivity.RELATIVE)
     @get:InputFiles
-    internal val editorConfigFiles: FileCollection = getEditorConfigFiles(project)
+    internal val editorConfigFiles: FileCollection by lazy(LazyThreadSafetyMode.NONE) {
+        getEditorConfigFiles(project)
+    }
```
Comment (JLLeitschuh) [inline plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt]: Can you leave a comment explaining why this is intentionally `lazy`? Other than that, this PR looks good!

Comment (Tapchicoma) [reply plugin/src/main/kotlin/org/jlleitschuh/gradle/ktlint/KtlintCheckTask.kt]: Added comment and force pushed.

**Pull request merged**